lib/config.js

```diff
@@ -23,12 +26,18 @@ module.exports.getConfig = async function getConfig({
   context,
   getConfig: fetchConfig
 }) {
-  const defaults = {
-    ...DEFAULT_CONFIG,
-    branches: context.payload.repository.default_branch
-  }
+  const repoConfig = await fetchConfig(
+    context,
+    CONFIG_NAME,
+    DEFAULT_CONFIG
+  ).catch(error => {
+    throw error
+  })
+
+  schema(context).validate(repoConfig)
 
-  const config = await fetchConfig(context, CONFIG_NAME, defaults)
+  const { error, value } = schema(context).validate(repoConfig)
+  const config = value
 
   try {
     config.replacers = validateReplacers({
```

Comment:
I'd like to explore Joi in a separate PR to see if it can handle the case for us.